Sort batch operation responses by actor type or actor id

korthout opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->

Adds the ability to batch operation responses by `actorType` or `actorId`. Both ascending and descending order are supported.

Note that no special care is taken to deal with `null` values. But, we'll consider this in the IT that will be added with:
- https://github.com/camunda/camunda/issues/42070

## Related issues

closes #42068

korthout committed:
test: sort batch operations by start date

Adds a test case that verifies that the rest gateway supports sorting
batch operations by startDate. The test setup is build in a way that
easily allows expanding to other sorting fields.

korthout committed:
test: sort batch operations by actorId

Adds a test case that verifies that the gateway can sort batch
operations by actorId. Note that the test fails because this is not yet
supported.

korthout committed:
feat: sort batch operations by actorId

Enables sorting batch operations by actor id.

korthout committed:
test: sort batch operations by actorType

Similar to actorId by now by actorType. Note that the test fails while
this is not implemented yet.

korthout committed:
feat: sort batch operations by actorType

korthout committed:
test: transform batch operation actor sort fields

Add test cases for sorting by actorType and actorId. Note these tests
fail right now.

korthout committed:
feat: transform actor sorting fields

I've placed the ACTOR_TYPE and ACTOR_ID fields under the new fields for
engine batch operations, as they belong there.

mustafadagher commented on zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/BatchOperationControllerTest.java:
```diff
@@ -323,6 +323,89 @@ void shouldResumeBatchOperation() {
+
+  @ParameterizedTest
+  @MethodSource("provideSortParameters")
+  void shouldSearchBatchOperationsSorted(
```
❓ I'm curious what are we testing here exactly? just that we accept the fields? (since the search request is mocked)

for example, in a similar example above to verify we pass the correct filter we did.
``` java
verify(batchOperationServices).search(new BatchOperationQuery.Builder().filter(filter).build());
```

🔧 would it be good to follow a similar pattern to verify the correct `sort` are passed down to the service layer?

mustafadagher reviewed (approved):
Thanks @korthout ! Looks good to me. 👍 
🔧 I have one suggestion about the test case, please consider it.

korthout committed:
test: assert sorting is passed along correctly

This test was not asserting that the implementation needs to pass the
sorting to the search layer. So it was relying only on the mocked
values and testing those.

korthout committed:
refactor: reuse sort object in test

Rather than specifying each sorting twice, we can simply use the
BatchOperationSort object to determine what we're doing.
